internal/deploy/compose.go

```diff
@@ -48,17 +48,27 @@ func (sm *DockerServiceManager) AddServicesToCompose(ctx context.Context, profil
 		"services": services,
 	}).Trace("Adding services to compose")
 
+	scaleCmds := []string{}
 	newServices := []ServiceRequest{profile}
 	for _, srv := range services {
 		newServices = append(newServices, srv)
+		if srv.Scale > 1 {
+			scaleCmds = append(scaleCmds, fmt.Sprintf("%s=%d", srv.Name, srv.Scale))
+		}
 	}
 
 	persistedEnv := state.Recover(profile.Name+"-profile", config.Op.Workspace)
 	for k, v := range env {
 		persistedEnv[k] = v
 	}
 
-	err := executeCompose(true, newServices, []string{"up", "-d"}, persistedEnv)
+	cmds := []string{"up", "-d"}
+	if len(scaleCmds) > 0 {
+		cmds = append(cmds, "--scale")
```

Comment:
Nice!!

Reply:
Yeah, I verified that the scale option scales the base image, not the current state of the service. OTOH, I did not include the k8s version to discuss it with you first